FIAT/finite_element.py

```diff
@@ -242,3 +247,67 @@ def entity_support_dofs(elem, entity_dim):
 
     cache[entity_dim] = result
     return result
+
+
+def _nhk(n, k):
+    """Choose k from n allowing for repetition"""
+    return numpy.math.factorial(n + k - 1) // numpy.math.factorial(n - 1) // numpy.math.factorial(k)
+
+
+def _set_barycentric_lattice_coordinates(a, i, j, m):
+    if m == 0:
+        return
+    asub = a[:, j]
+    i0 = i
+    if j == 0:
+        asub[i0: i0 + 1] = m
+        return
+    for v in range(m + 1):
+        i1 = i0 + _nhk(j, m - v)
+        asub[i0: i1] = v
+        _set_barycentric_lattice_coordinates(a, i0, j - 1, m - v)
+        i0 = i1
```

Comment:
A bit hard to follow...

Reply:
These weird bits have all been removed. Thanks!